core/embed/rust/src/ui/model_mercury/component/swipe_up_screen.rs

```diff
@@ -0,0 +1,69 @@
+use crate::ui::{
+    component::{Component, Event, EventCtx, PageMsg},
+    geometry::Rect,
+    shape::Renderer,
+};
+
+use super::{Swipe, SwipeDirection};
+
+/// Wrapper component adding "swipe up" handling to `content`.
+pub struct SwipeUpScreen<T> {
+    content: T,
+    swipe: Swipe,
+}
+
+impl<T> SwipeUpScreen<T>
+where
+    T: Component,
+{
+    pub fn new(content: T) -> Self {
+        Self {
+            content,
+            swipe: Swipe::new().up(),
+        }
+    }
+}
+
+impl<T> Component for SwipeUpScreen<T>
+where
+    T: Component,
+{
+    type Msg = PageMsg<T::Msg>;
```

Comment:
SwipeUpScreen makes sense to me, in fact with the T3T1 animations we should probably stop using `struct Swipe` in most places, but that is out of scope for this PR.

As for the message type, let's use `type Msg = ();` or an enum `{ Swiped, Content(inner) }` if needed.

Reply:
Ok, I did go with the enum
https://github.com/trezor/trezor-firmware/pull/3740/commits/c0f16354c5012e6804ac9ce36819ff349082f61a